crates/next-api/src/app.rs

```diff
@@ -880,21 +880,33 @@ impl AppEndpoint {
                     None
                 };
 
-                let client_dynamic_imports = collect_next_dynamic_imports(
-                    client_references
-                        .await?
-                        .client_references
-                        .iter()
-                        .filter_map(|r| match r.ty() {
-                            ClientReferenceType::EcmascriptClientReference(entry) => Some(entry),
-                            ClientReferenceType::CssClientReference(_) => None,
-                        })
-                        .map(|entry| async move { Ok(Vc::upcast(entry.await?.ssr_module)) })
-                        .try_join()
-                        .await?,
-                    Vc::upcast(this.app_project.client_module_context()),
-                )
-                .await?;
+                let mut client_references_by_server_comp = IndexMap::new();
```

Comment:
This map might look like this
```
[
    (
        "[project]/bench/heavy-npm-deps/app/layout.js [app-rsc] (ecmascript, Next.js server component)",
        [
            "[project]/bench/heavy-npm-deps/components/lodash.js [app-ssr] (ecmascript)",
        ],
    ),
    (
        None,
        [
            "[project]/packages/next/dist/client/components/error-boundary.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/layout-router.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/render-from-template-context.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/client-page.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/client-segment.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/not-found-boundary.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/error-boundary.js [app-ssr] (ecmascript)",
        ],
    ),
]
```
and for another page:
```
[
    (
        "[project]/bench/heavy-npm-deps/app/layout.js [app-rsc] (ecmascript, Next.js server component)",
        [
            "[project]/bench/heavy-npm-deps/components/lodash.js [app-ssr] (ecmascript)",
        ],
    ),
    (
        "[project]/bench/heavy-npm-deps/app/sub2/page.js [app-rsc] (ecmascript, Next.js server component)",
        [
            "[project]/bench/heavy-npm-deps/components/mermaid.js [app-ssr] (ecmascript)",
        ],
    ),
    (
        None,
        [
            "[project]/packages/next/dist/client/components/error-boundary.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/layout-router.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/render-from-template-context.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/client-page.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/client-segment.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/not-found-boundary.js [app-ssr] (ecmascript)",
            "[project]/packages/next/dist/esm/client/components/error-boundary.js [app-ssr] (ecmascript)",
        ],
    ),
]
```

Reply:
I think it would make sense to have the `None` server component key first, since they are shared between everything.

Calling this before the loop would do that:

``` rs
client_references_by_server_comp.insert(None, Vec::new())
```